### ISSUE TITLE
Sidebar categories linking to a doc or category index page

### ISSUE THREAD

**Issue by XavierNV** [facebook/docusaurus#2643]
## 🚀 Feature

Category in docs are only container for subitems (topics). But some help system enable category to be introductory topic to subitems (ie containing text that introduce the subitems or refers to subitems  ie see also links to subtopic1, sub..)

### Have you read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

## Motivation

Interested to use the Docusaurus Help System but before investing in it wan't to know the roadmap

## Pitch

It's pretty useful. If you can read the introductory topic, you sometimes don't have to click/navigate to subtopics. It'a a better system than having a first topic (~overview) refering to following topics in a same level appearance > introductory topic has to be a level upper the subtopics..
Example of documentation with introductory topics :
https://www.gatsbyjs.org/docs/deploying-and-hosting/
https://www.innovasys.com/help/hs2019.1/authoringcontent.html
https://markdownmonster.west-wind.com/docs/

Hope my explaination is not too hard to understand..
Thanks to the docusaurus team for their work

Regards, Xavier

**Comment by hsiaoer** [comment]
+1 to this feature request. I have a case right now where I'm migrating docs over to docusaurus and we've set all nav items to be expanded by default.

```
themeConfig: {
    sidebarCollapsible: false,
}
```

Currently if you try clicking on the category item in the nav, nothing happens and it would be helpful to be able to add an optional link to a doc page on the category:

Example where introDoc.md is the doc when you click on the nav item.
```
{
        type: "category",
        label: "Introduction",
        rootItem: introDoc,
        items: [subdoc1, subdoc2]
}
```

**Comment by yangshun** [comment]
Some users want this behavior but some other users prefer that the category are just labels and not clickable. If we were to add support for this, we might need to change how the pure categories look, such that they don't look clickable.

**Comment by sladyn98** [comment]
@yangshun Is this still a starter issue to be worked on if yes I could maybe give it a shot.

**Comment by slorber** [comment]
Hey, didn't notice this issue, that also seems related to a more recent one: https://github.com/facebook/docusaurus/issues/3051

@sladyn98 there's ongoing work to be merged on the docs plugin + a few other important refactors I'd like to do.
This won't be time lost to have such a POC of this feature, and would help collect feedback, but just be aware that you might not be able to rebase it easily, and might need to re-open a brand new pull request 🤪

**Comment by slorber** [comment]
Raw thoughts but here's a possible api:


```
{
        type: "categoryDoc",
        label: "Introduction",
        id: "introDocId",
        items: [subdoc1, subdoc2]
}
```

Would render the introDocId doc.

```
{
        type: "categoryDoc",
        label: "Introduction",
        id: undefined, // on purpose
        items: [subdoc1, subdoc2]
}
```

Would render an automatically generated table of content of all the inner-tree of this category (we could later see how to make this view more useful by adding additional data to it, maybe a small doc summary or something)

**Comment by TomPeirs** [comment]
Is there any update on this feature?
Is there any proposed solution for now?
Currently I have a duplicate entry in my sidebar representing the category file. This of course looks a little bit ugly.

**Comment by slorber** [comment]
This is quite often requested (https://github.com/facebook/docusaurus/issues/3686) so we'll try to find a solution, but it will come after i18n support.

In the meantime, if someone wants to design the API surface and write an exhaustive RFC, so that it is flexible and suit all the use-cases reported, that would help. 

The design of this feature is not so straightforward as we are not sure yet how it should behave regarding category collapsing.

**Comment by chrisdostert** [comment]
Just wanted to add that this is how the [kubernetes.io](https://kubernetes.io/docs/concepts/overview/) docs work which to me are great. Would love to have this be available (if not default) behavior in docusaurus!

**Comment by ctlajoie** [comment]
I would definitely like to see this as well. I have encountered the requested functionality several times just today on different sites. In addition to the [kubernetes.io docs](https://kubernetes.io/docs/concepts/overview/) mentioned by @chrisdostert:

- [GitHub docs](https://docs.github.com/en/free-pro-team@latest/packages/getting-started-with-github-container-registry)
- [DITA docs](https://www.dita-ot.org/3.5/topics/input-formats.html)

I like the sidebar behavior of the GitHub docs best (although the k8s docs operate similarly). It hides the sub-articles of the other categories that are not related to doc currently in view. I do not find this behavior confusing or annoying. It makes the sidebar easier to grok.

**Comment by slorber** [comment]
Thanks for showing some examples. We'll try to find an API that allows all these slightly different UX so that you can choose the one you want.

> I like the sidebar behavior of the GitHub docs best (although the k8s docs operate similarly). It hides the sub-articles of the other categories that are not related to doc currently in view. I do not find this behavior confusing or annoying. It makes the sidebar easier to grok.

At the same time it's confusing if the item you click on immediately change position. I guess it's an acceptable tradeoff, but we should probably not enforce this specific UX.

**Comment by slorber** [comment]
FYI there's an existing PR, and also suggested an API here, curious to have your feedback on it: https://github.com/facebook/docusaurus/pull/3898#issuecomment-742638270

**Comment by omry** [comment]
Thanks @slorber.
What I had in mind (which is in line with the examples provided by @ctlajoie, is to have a space to write an overview page for the category that would be linked to when clicking on the category itself.

The PR you linked to seems to be solving a different (if related) problem of how to minimize clicks in a world where you are forced to create an intro page as a subpage.

As a workaround it's a pretty clean one, but not exactly the request here.

**Comment by slorber** [comment]
@omry , you could create the overview md page, and make the category link to it?

```js
module.exports = {
    docs: [
    {
      type: 'category',
      label: 'Docusaurus',
      link: {type: "doc", id: 'docusaurus-category-overview-doc'}
      items: ['introduction', 'design-principles', 'contributing'],
    },
    // ...
  },
};
```

Do you see a good reason for this category overview page to not be a regular doc?

**Comment by omry** [comment]
I think it should be a regular doc in the sense that I should be able to author it like any other doc.
This is mostly about the user experience: As a user I don't think of the content of the high level overview of a category as something at the same level as the items of the category.

Can you show me an example of what your above overview page looks like?
I have been adding an item to the list as an alternative for an index page. e.g: 

![image](https://user-images.githubusercontent.com/376455/101820588-ac6cea80-3adb-11eb-8e03-863f96a11d41.png)

**Comment by nascode** [comment]
Hi, i want to open doc when user click on a category in the sidebar. Looks like it's not supported yet.

Thus i try to create workaround for it by using  first index of category item docs and modify sidebar component with sizzling.

You can see it on my repo https://github.com/nascode/example-docusaurus-v2-sidebar-category-doc

**Comment by lunchboxav** [comment]
that looks very helpful @nascode , thank you!

**Comment by slorber** [comment]
@omry what I understand is you want "tutorials" to link to the overview, instead of "Tutorials intro"

![image](https://user-images.githubusercontent.com/749374/101895305-a6444000-3ba7-11eb-868e-d5157e5d475e.png)


But isn't that exactly what I'm proposing to you?

```js
module.exports = {
    docs: [
    {
      type: 'category',
      label: 'Tutorials',
      link: {type: "doc", id: 'tutorials-intro'}
      items: ['basic-tutorial', 'structured-config-tutorial'],
    },
    // ...
  },
};
```

> Can you show me an example of what your above overview page looks like?

Exactly the same as your current intro page, it's just the sidebar that will be different

**Comment by omry** [comment]
I tried to modify my sidebar to this and was not successful (I kept getting parsing errors from the sidebar).
[Here is my sidebar](https://github.com/facebookresearch/hydra/blob/b2d5310a8a89bdaa8d9fe92b9fc0a90b3a2d5edb/website/sidebars.js#L13-L44), Can you help me get on the right track?

**Comment by slorber** [comment]
@omry the code above is not implemented yet, it's just a possible API design we could use to implement this feature.

**Comment by omry** [comment]
Oh, I thought you meant that this was already supported somehow.

I think we are on the same page on the functionality aspects but let me repeat it to be sure:
What I would like is for the Tutorial category, when clicked would do two things:
1. change the content of the right pane to that of the page "Tutorials intro".
2. Expand the categories list (to contain the items Basic Tutorial and Structured Configs Tutorial).

This should also work when linking to https://hydra.cc/docs/tutorials/ (currently it results in 404).

Your proposed API design seems sensible to me.

In terms of API design for this, the dict describing the tutorials category above looks good to me.

I am assuming you are aware that currently the top level doc is a dict and not a list for some people (all?) as evident by my sidebar linked above and your suggestion is either a mistake or some other change that is planned:

my sidebar:
```
module.exports = {
    docs: {
       ...
    }
```

Your proposal:
```
module.exports = {
    docs: [
       ...
    ]
}
```

**Comment by TomPeirs** [comment]
What I think would be the desired effect is to simply be able to link a category to a markdown file itself
See example https://www.gatsbyjs.com/docs/recipes/

The following comment would not enable that as you would autogenerate some content, but it would be nice if we could define ourselves what to write on the intro/overview page of the cateogry
https://github.com/facebook/docusaurus/issues/2643#issuecomment-661058279

As mentioned by @hsiaoer , it would be helpful to be able to add an optional link to a doc page on the category:
Example where introDoc.md is the doc when you click on the nav item.
`{
        type: "category",
        label: "Introduction",
        rootItem: introDoc,
        items: [subdoc1, subdoc2]
}`

**Comment by slorber** [comment]
@omry yes we agree

> This should also work when linking to https://hydra.cc/docs/tutorials/ (currently it results in 404).

You can already use `slug: /docs/tutorials` on your existing category intro doc for that

> I am assuming you are aware that currently the top level doc is a dict and not a list for some people

As documented, there are 2 syntaxes (shorthand/dictionary, expanded/list) but they are equivalent, it's just a shortcut we created. Also you'll have to use the expanded form if you want more options like the one we are discussing here, as the shorthand is just `categoryName  -> subItems` and can't pass more config


---

@TomPeirs 

> What I think would be the desired effect is to simply be able to link a category to a markdown file itself
> See example https://www.gatsbyjs.com/docs/recipes/

Yes, this is exactly what my API description above would solve

> The following #2643 would not enable that as you would autogenerate some content, but it would be nice if we could define ourselves what to write on the intro/overview page of the category

I don't understand, it's what I suggest  to do: we assign a docId to a category instead of generating automatically the category page when we write such config:

![image](https://user-images.githubusercontent.com/749374/102201904-fc2a2800-3ec6-11eb-8781-2a12e36d9dd6.png)

It's your responsibility to write an intro/category doc with `id: tutorials-intro` here, we won't auto-generate it.

This is very similar to what you suggest, except we didn't  use the same name/api surface (`rootItem: "id"` replaced by `link: {type: "doc", id: "id"}}` for extensibility)

**Comment by omry** [comment]
Thanks @slorber 
> You can already use slug: /docs/tutorials on your existing category intro doc for that

1. Can you point to an example?
2. This will still be needed once you support sidebar category doc?

**Comment by slorber** [comment]
The example is just what I gave you, just try this frontmatter and see how
it impacts your doc url
https://v2.docusaurus.io/docs/docs-introduction/


The presence of a doc in a sidebar (either as a regular doc or a caterory
doc) does not affect its url in anyway. Only the slug/id/filename do.

You are free to choose the exact url you want for all your docs using the
slug frontmatter

Le ven. 18 déc. 2020 à 04:31, Omry Yadan <notifications@github.com> a
écrit :

**Comment by slorber** [comment]
Came across this Hashnode doc homepage: https://support.hashnode.com/

I think this kind of UI can be nice for the auto-generated category index:

![image](https://user-images.githubusercontent.com/749374/106868855-95a7d600-66cf-11eb-988e-7b8d474edd1e.png)

If there are nested items on a 3rd level we could add a `(count)` text and if there are too many level2 items we could add some kind of accordion

**Comment by martis-git** [comment]
Styill actual!!

**Comment by leadream** [comment]
What‘s the progress? Still waiting for it.

**Comment by slorber** [comment]
This feature is one of the top priorities in my todo-list. 
I expect it to be implemented in October.

**Comment by Zenahr** [comment]
@slorber Is there a working (but not yet released/merged) version of this feature that we could source ourselves at the danger of incompatibility with future changes? (just as a temporary solution until it's been merged.

**Comment by slorber** [comment]
There is no easy way to implement this in userland that I can think of. 

You can try to assign extra data in the `customProps` attribute of sidebar items, and provide your own rendering logic using those custom props.
https://docusaurus.io/docs/sidebar#passing-custom-props

**Comment by Josh-Cena** [comment]
Gitbook makes the `README.md` file within a folder the category index page, which I think we can adopt for auto-generated sidebars as a default value

**Comment by damageboy** [comment]
When editing content heavy websites, I think it would be useful to link automatically to foldr-name.md

This makes the editing process in IDEs such as vs-code much nicer by having `folder1.md`, `folder2.md` etc. open side by side rather than 10 tabs with `README.md`

**Comment by Josh-Cena** [comment]
> This makes the editing process in IDEs such as vs-code much nicer by having `folder1.md`, `folder2.md` etc. open side by side rather than 10 tabs with `README.md`

Maybe in the end we shouldn't adopt any default value and just use front matter like `category_index: true` to turn a page into the index page?

**Comment by slorber** [comment]
Working POC: https://deploy-preview-5830--docusaurus-2.netlify.app/tests/docs/sidebar/category/tests/
There are some edge-cases not covered yet, and no tests, but it works.

The link:
- is a generated category index page (rough styling)
- contains categories linking to actual docs using `index.md`, `readme.md` or `<categoryName>.md` convention

The goal for me is to make it easy for you to provide custom styling of the generated category index page, for which you could use the sidebar `customProps` attributes to provide custom stylings like colors etc... We'll try to do that on the Docusaurus website as an example and to make our site beautiful.

Note a UX edge case that must be considered: some sub-categories of the generated category index won't always have links (ie they'll be like current categories, appearing in white here):

![image](https://user-images.githubusercontent.com/749374/140562654-e13acff7-7e4d-451c-b9ff-f89d84f70f28.png)

**Comment by Josh-Cena** [comment]
My two cents

> should we render lists or cards

Cards. The list is too plain to be attractive & meaningful

> should we show the full tree render the direct subitems of the category

Only the direct children. Deeper children can be accessed in other ways

> should we render collapsed categories

If we ended up rendering the whole tree, we should render collapsed categories. The default collapsed state is very arbitrary and shouldn't interfere with the statically rendered page.

On how the category link should appear in the sidebar, I prefer

- Only highlight text on hover if the category is a link
- Clicking on the triangle toggles the category without redirecting
- Clicking on the text toggles the category & redirects

**Comment by slorber** [comment]
> Cards. The list is too plain to be attractive & meaningful

Agree

> Only the direct children. Deeper children can be accessed in other ways

Agree


---

Links to test:

deploy-preview-5830--docusaurus-2.netlify.app/docs/next/category/getting-started
deploy-preview-5830--docusaurus-2.netlify.app/docs/next/category/guides
deploy-preview-5830--docusaurus-2.netlify.app/tests/docs/category/tests


This is not the most beautiful design but is a good first version, what should we change?

What should the cards and layout look like?

Any SVG icons to recommend instead of emojis?

![image](https://user-images.githubusercontent.com/749374/141179313-fffcd577-ef93-42fd-b977-85f34e3082a7.png)

**Comment by Josh-Cena** [comment]
I have these two types of design to offer:

- https://docs.luxalgo.com/ (ignore the box shadow part, just the spacing / size etc.)
- https://docs.luxalgo.com/lux-algo-premium-indicators/lux-algo#features

I personally would prefer smaller font size & larger padding. Maybe we can also render some description and/or the pages' routes?

**Comment by dsmmcken** [comment]
I would use this feature. I currently have a page where I have an intro blurb + cards (with title+description) for a section. I just have it as the first item, but I would prefer it as link on the category itself.

Ideally, auto-gen is great, but personally I probably would want to be able to point a category to my own md file for each category where desired, so I can add some text above or whatever, and easily be able to access the data of docs in that category to make my own cards.

ex. https://deephaven.io/core/docs/tutorials/overview/

I was inspired by the gatsby docs:
https://www.gatsbyjs.com/docs/how-to/performance/

**Comment by dsmmcken** [comment]
https://deephaven.io/core/docs/tutorials/overview/ is the final result I would like, and what I've got currently. I would like it as the category link, rather then the first item under the category, and an easier way to get the data for the cards.

Something to consider with cards, is maybe including the description from the front-matter as well on the card.

**Comment by slorber** [comment]
@dsmmcken it looks quite complicated to achieve what you want here because your cards are not really following a "normalized structure".

This top part could be achieved by adding some category metadata:

![image](https://user-images.githubusercontent.com/749374/142736797-bcf1f820-5a23-4320-85e6-d8aca62118e0.png)

---

The bottom part is more complex.

![image](https://user-images.githubusercontent.com/749374/142736831-0f3018a3-761f-43cd-a3f4-035ca76a81ff.png)

The first card leads to a doc, others are liking to headings/anchors... At some point, Docusaurus should be able to understand what it needs to create a card for.

What we'll do automatically (show cards for subcategory items, not headings) might not make sense in all use-cases.

---

But in the end, we should enable you to build your own index rendering logic and replace our own default logic.

With swizzle, you could be able to build your own rendering logic and create a sidebar category with a custom proprietary syntax, such as:

```js
module.exports = {
  mySidebar: [
    {
      type: "category",
      label: "Tutorials",
      items: ["..."],
      link: { type: "generated-index" },
      customProps: {
        title: "Deephaven Tutorial",
        description: "Learn how Deephaven works",
        subtitle:
          "The goal of this tutorial is to help you create a mental model for how Deephaven works and to introduce you to its capabilities and workflows.",
        intro:
          "You will build analytics and applications with both updating (streaming) and static (batch) data sets, become familiar with the Deephaven IDE experience, apply your Python functions to your data, build a dashboard, create a notebook, and build and launch a stand-alone application. With a bit more time, you’ll then develop a Python client application that uses the API to receive data from your server application.",
        cards: [
          {
            title: "0. Quick Start",
            description:
              "Set up Deephaven in your Docker environment with pre-built images.",
            url: "/core/docs/tutorials/quickstart/"
          },
          {
            title: "1. Ingest static data",
            description:
              "Import CSVs from URLs. Read an example CSV and Parquet file..",
            url: "/core/docs/tutorials/tutorial/#1-ingest-static-data"
          }
          // ... other cards here
        ]
      }
    }
  ]
};
```

Does it make sense? 

Not sure it would be significantly simpler than linking to a custom doc though (or using a `index.md` or `README.md` which we'll support).

**Comment by slorber** [comment]
If you have any feedback to share, on what you see here, now is a good time:

- Category linking to doc: https://deploy-preview-5830--docusaurus-2.netlify.app/docs/next/api/plugins/
- Category linking to automatically generated index: https://deploy-preview-5830--docusaurus-2.netlify.app/docs/next/category/guides

I'm in the process of making this robust, tested and documented, so let me know if you have any UX feedback

**Comment by dsmmcken** [comment]
That looks pretty good. I think linking to a page, or auto generated will work for most people.

Something I find awkward is not being able to collapse an opened category. If I open a category in the middle and it's giant, then I want a doc in the category after, it is difficult to have to scroll way past it. Should you still be able to collapse a category if it's active, by clicking it? Maybe the icon can performs that action? I don't know, but that bit feels off.

My page having weird links to headings is actually just a temporary solution, we intend them to be separate pages (our code snippets are auto extracted, tested, run, and then automatically inject their output back into the page, but we haven't create the ability to have snippets depend on other pages yet).

If possible, it would be nice if you could also fetch the card data only somehow, where I could have an index page, but then still render cards on it as well.

some-category-doc.md
```
---
etc
---

my custom intro or whatever markup here

<MyCustomCardComponent cards={some-magic-way-to-have-card-data-here} />
```

Kinda like how you can add a <Toc /> component to a page.

**Comment by Josh-Cena** [comment]
> Something I find awkward is not being able to collapse an opened category.

Yep, it's actually blocked by an upstream dep: https://github.com/facebookincubator/infima/issues/188 (it's also developed by us anyways, so not so much trouble, just that the fix isn't in place yet)

> If possible, it would be nice if you could also fetch the card data only somehow, where I could have an index page, but then still render cards on it as well.

+1. And I do have an idea of how to implement this, let's see if we want to add that in this PR or raise a subsequent PR because this one is getting big

**Comment by TomPeirs** [comment]
Hi @slorber  and @Josh-Cena 
Thank you for working on this interesting feature.

My input would be to make the cards render similar to an admonition, as I find contrast-wise they seem to adhere more to W3C accessibility standards to me.
The hover effect and the cursor handling is nice, but the border shadow hurts my head a little.

**Comment by Josh-Cena** [comment]
> My input would be to make the cards render similar to an admonition, as I find contrast-wise they seem to adhere more to W3C accessibility standards to me.

I guess you are talking about the same problem as the box shadow. See https://github.com/facebookincubator/infima/issues/187. I don't think the current design is problematic but certainly we can sharpen the shadow when not hovered.

**Comment by TomPeirs** [comment]
> > My input would be to make the cards render similar to an admonition, as I find contrast-wise they seem to adhere more to W3C accessibility standards to me.
> 
> I guess you are talking about the same problem as the box shadow. See [facebookincubator/infima#187](https://github.com/facebookincubator/infima/issues/187). I don't think the current design is problematic but certainly we can sharpen the shadow when not hovered.

Indeed, that would work.

**Comment by TomPeirs** [comment]
@Josh-Cena @slorber 
One more thing to consider for Card Icons.
The sidebar object can take the following : doc, ref, link, category.
In the netlify preview we can see the icons for a doc and a category, and I was wondering if you considered also Link?

Perhaps we could add a **little** bit more spacing between the Title and the Icon of a card?

Would there be a possibility to add a customization option to assign those icons/svgs for the card icons?

**Comment by Josh-Cena** [comment]
> Perhaps we could add a little bit more spacing between the Title and the Icon of a card?

I personally think the spacing is good...

> Would there be a possibility to add a customization option to assign those icons/svgs for the card icons?

If you have differing design ideas you can likely re-implement your own index page through swizzling. (And we will try to make it as convenient as possible.) We would likely not add options for every detail.

> In the netlify preview we can see the icons for a doc and a category, and I was wondering if you considered also Link?

Indeed, links, unsurprisingly, have the icon 🔗.

**Comment by TomPeirs** [comment]
> > Perhaps we could add a little bit more spacing between the Title and the Icon of a card?
> 
> I personally think the spacing is good...
> 
> > Would there be a possibility to add a customization option to assign those icons/svgs for the card icons?
> 
> If you have differing design ideas you can likely re-implement your own index page through swizzling. (And we will try to make it as convenient as possible.) We would likely not add options for every detail.
> 
> > In the netlify preview we can see the icons for a doc and a category, and I was wondering if you considered also Link?
> 
> Indeed, links, unsurprisingly, have the icon 🔗.

Oke thank you ;-) seems that everything is considered.

**Comment by slorber** [comment]
Yes, we'll allow collapsing a category

External links have a different icon, this can be seen here: https://deploy-preview-5830--docusaurus-2.netlify.app/tests/docs/category/tests/

> If possible, it would be nice if you could also fetch the card data only somehow, where I could have an index page, but then still render cards on it as well.

I don't understand what this means sorry 😅 

> Perhaps we could add a little bit more spacing between the Title and the Icon of a card?
> Would there be a possibility to add a customization option to assign those icons/svgs for the card icons?

We'll likely improve the design and add customization options later. Here I'm mostly focusing on the initial MVP with the UX and a good extensible public API that we can improve later. 

Note that these theme components are well encapsulated and easy to customize with a swizzle.

--- 

Thanks for your feedback, which is mostly about design and "shortcut" options, we can improve all that incrementally from the current state later in subsequent PRs, and in the meantime using swizzle is likely a good enough workaround until we figure out the most wanted customization apis.

**Comment by Josh-Cena** [comment]
> If possible, it would be nice if you could also fetch the card data only somehow, where I could have an index page, but then still render cards on it as well.

@slorber By that he means, you have a normal doc page as index page, but inject the category items in it as JSON so it can be rendered somewhere in the middle of the page, kind of like the `TOCInline` where we have a global `toc` export to do this.

My idea would be to inject this in `MDXProvider`

**Comment by slorber** [comment]
Oh I see, something like this?

![image](https://user-images.githubusercontent.com/749374/143593769-51d00b08-f0a8-46a9-94c2-1a33c7eec29d.png)

It's already technically possible atm using hooks in the MDX doc, but probably not with the most intuitive API. We can figure out later how to add MDX shortcodes for that.

**Comment by slorber** [comment]
PR and UX ready for a final review!

Unless there's any pushback, this is going to be merged tomorrow!

https://github.com/facebook/docusaurus/pull/5830

![image](https://user-images.githubusercontent.com/749374/144483045-4b390493-dbdc-4f6c-951e-47ed6499df90.png)

![image](https://user-images.githubusercontent.com/749374/144483072-9cbac90f-2482-4fc6-b1bf-4c5c2dd1e5a2.png)


## Test URLs:

- https://deploy-preview-5830--docusaurus-2.netlify.app/docs/next/category/getting-started
- https://deploy-preview-5830--docusaurus-2.netlify.app/docs/next/category/guides
- https://deploy-preview-5830--docusaurus-2.netlify.app/docs/next/markdown-features/
- https://deploy-preview-5830--docusaurus-2.netlify.app/tests/docs/category/tests


## Docs:

- [Category links (doc, generated-index)](https://deploy-preview-5830--docusaurus-2.netlify.app/docs/next/sidebar/#category-link)
- [Category `index.md` convention](https://deploy-preview-5830--docusaurus-2.netlify.app/docs/next/sidebar/#category-index-convention)
- [docCategoryGeneratedIndexComponent option](https://deploy-preview-5830--docusaurus-2.netlify.app/docs/next/api/plugins/@docusaurus/plugin-content-docs/#docCategoryGeneratedIndexComponent)

---

Please share your last feedback asap

**Comment by dsmmcken** [comment]
Is it easy to change/remove the icons, or would I have to swizzle the component for that?

**Comment by Josh-Cena** [comment]
> Is it easy to change/remove the icons, or would I have to swizzle the component for that?

@slorber Is it possible to popularize the idea of CSS `::before` pseudoelements for these icons?

**Comment by seanbrakefield** [comment]
I’ve been following the progress of this for a while now, and I love how this is forming. Really great work!

Request:
Can you close the Table of Contents panel when tapping the Category item (that has a page) on the mobile version? Currently, that panel stays open which makes it seem like nothing has occurred.

**Comment by slorber** [comment]
> Is it easy to change/remove the icons, or would I have to swizzle the component for that?

> @slorber Is it possible to popularize the idea of CSS ::before pseudoelements for these icons?

For now you can swizzle the `DocCard` component

We don't have yet a good way to manage theme icons, we'll try to normalize how we handle icons in themes as part of https://github.com/facebook/docusaurus/issues/5865 and make sure that overriding an icon is consistent for all the existing use-cases of the theme.

If you have a better icon/design we can also improve our current design. Feel free to override `DocCard` in userland for now, show us the result, and eventually if the design is better we can make it the default later.

---

> Can you close the Table of Contents panel when tapping the Category item (that has a page) on the mobile version? Currently, that panel stays open which makes it seem like nothing has occurred.

Yes, also noticed that ;) this has been fixed

**Comment by slorber** [comment]
This has been published in canary (0.0.0-4241) and scheduled for a release next week

Please give it a try and tell us if it works fine ;)

**Comment by SivertGullbergHansen** [comment]
Hey guys 😄 
I've been struggling to find the option to enable this? 
I didn't want to bother reading through all these comments, therefore I am asking :)

I also couldn't find anything related to it in your docs

**Comment by Josh-Cena** [comment]
@SivertGullbergHansen

- https://docusaurus.io/docs/sidebar/items#category-link
- https://docusaurus.io/docs/sidebar/autogenerated#category-item-metadata

**Comment by bfelbo** [comment]
> If you can read the introductory topic, you sometimes don't have to click/navigate to subtopics. It'a a better system than having a first topic (~overview) refering to following topics in a same level appearance > introductory topic has to be a level upper the subtopics.. Example of documentation with introductory topics : https://www.gatsbyjs.org/docs/deploying-and-hosting/

Interestingly, Gatsby actually moved away from this style. My guess would be that it's because readers might miss that they can click on the category sidebar to read info. Readers might sometimes expand the category using the arrow instead, thereby missing crucial introductory info. This can also happen if the reader opens a category subpage from an external link, in which case it's unlikely they'll know to click the category in the sidebar since it's also expanded.

I'd argue that a better approach would be to make any clicks to the category redirect the reader to the intro subpage. That way, it's still a quick experience (same # of clicks), but makes the intro information explicit. I've created https://github.com/facebook/docusaurus/issues/8833 to add support for this :)
